Upgrade yarn version to v3

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
https://github.com/grapl-security/issue-tracker/issues/844
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
Upgrades the engagement_view to use yarn v3.
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->
We're upgrading to yarn v3 so that
a. we fix an issue we're seeing in CI.
b. we're on the latest and have a path to enabling zero installs and
other features.
c. Get the yarn versions of local developement and the docker container
to match
### How were these changes tested?
Automated test suite, manual testing
<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Upgrade yarn version to v3

We're upgrading to yarn v3 so that
a. we fix an issue we're seeing in CI.
b. we're on the latest and have a path to enabling zero installs and
other features.
c. Get the yarn versions of local developement and the docker container
to match

inickles-grapl commented on src/js/engagement_view/.yarnrc.yml:
```diff
@@ -10,4 +8,4 @@ installStatePath: .yarn/state/install-state.gz
 nodeLinker: node-modules
 
-yarnPath: .yarn/releases/yarn-berry.cjs
+yarnPath: .yarn/releases/yarn-3.1.1.cjs
```
I think you want to just remove this line as well since it's no longer being sourced from that location.

twunderlich-grapl replied:
So while true that the docker container has the yarn version set correctly, anyone running yarn locally won't necessarily be using the right yarn version. So far, this seems to be the preferred way to set the version although I do agree its stupid to basically have this twice. 
If you know of a config option I've missed to force local development to use the right yarn version please let me know since its been a while since I've done serious yarn spelunking

Commit:
Fix hadolint check for buildenv.dockerfile

This combines the two RUN statements per hadolint's recommendation. If
it turns out this causes cache busting, we can split it up again.
As part of this, I've updated the in-line documentation.

Commit:
Commit the yarn release file

This is required so that tests pass in CI

Commit:
Remove yarn release file and yarnpath to it

Now that we're setting the yarn version explicitly in the Dockerfile
we don't need to keep a yarn binary around. This removes it and removes
the reference to it.
Keep in mind that this may break local development if anyone is doing
that since we're only specifying the version in the Dockerfile
